lib/net.js

```diff
@@ -660,10 +660,12 @@ Socket.prototype._destroy = function(exception, cb) {
     this._handle.onread = noop;
     this._handle = null;
     this._sockname = null;
+    cb(exception);
+  } else {
+    cb(exception);
+    process.nextTick(emitCloseNT, this);
   }
 
-  cb(exception);
```

Comment:
Does this moving this into the individual branches of the `if` matter, if `emitCloseNT` is delayed by a tick anyway?

Reply:
@addaleax: yes, if the user wants to schedule something before `'close'` using a `nextTick`.

Reply:
I think I'm very close the edge on the edge case on this one :)